x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -1,27 +1,52 @@
 lexer grammar EsqlBaseLexer;
 
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+@header {
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+}
+
+options {
+  superClass=LexerConfig;
+  caseInsensitive=true;
```

Comment:
ANTLR supports caseInsensitivity per whole file or per statement - this works across languages and simplifies the grammar (no need for a caseinsensitive stream wrapper).
This is unrelated to the core PR but it simplifies its wiring.

Reply:
We have actually been [injecting this](https://github.com/elastic/kibana/blob/main/.buildkite/scripts/steps/esql_grammar_sync.sh#L20-L22) into the grammar at the Kibana level. Nice to align.

Reply:
Does this mean named parameter's name will be case insensitive?

Reply:
The behavior is similar to the previous `CaseChangingCharStream` namely the lexer sees the tokens with the same case, regardless of declaration. However the underlying text (especially values and identifiers) is NOT changed.